testutil/network.go

```diff
@@ -14,29 +14,23 @@ import (
 )
 
 func checkPortIsOpen(port int) bool {
-	timeout := time.Millisecond * 10
+	timeout := time.Second
 	conn, err := net.DialTimeout("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(port)), timeout)
 	if err != nil {
-		// port not opened
 		return false
 	}
-	// port has opened
-	if conn != nil {
-		conn.Close()
-	}
+	defer conn.Close()
 	return true
 }
 
 // RandomPort returns a random port number between 30000 and 50000
 func RandomPort() int {
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-	port := r.Intn(2000) + 30000
-	for i := 0; i < 18000; i++ {
-		if checkPortIsOpen(port) == false {
-			break
+	portStart, portEnd := r.Intn(2000)+30000, 50000
+	for port := portStart; port < portEnd; port++ {
+		if checkPortIsOpen(port) {
+			return port
 		}
-		port++
-		// retry next port
 	}
-	return port
+	return -1
```

Comment:
if we change to return `-1` here,
then need to check all places using `RandomPort()` against `-1`?

Reply:
All places using RandomPort() are in the test files. I assume it should succeed to generate a random number.
There are two version implementation for this function. They return different result when no ports is opening between 30000 and 50000. Which implementation you suggest?
1. The first version (old version) return 50000
```
func RandomPort() int {
	r := rand.New(rand.NewSource(time.Now().UnixNano()))
	var port int;
	for port = r.Intn(2000)+30000; port < 50000; port++ {
		if checkPortIsOpen(port) {
			break;
		}
	}
	return port
}
```
2. The second version return -1
```
func RandomPort() int {
	r := rand.New(rand.NewSource(time.Now().UnixNano()))
	portStart, portEnd := r.Intn(2000)+30000, 50000
	for port := portStart; port < portEnd; port++ {
		if checkPortIsOpen(port) {
			return port
		}
	}
	return -1
}
```

Reply:
i see, since we don't check the returned value, i think using old version is better (50000 is a valid port number, but -1 is not)

Reply:
pls also rebase and fix test error

Reply:
test failed due to protobuf issue, i think we need to upgrade the protobuf in `iotex-proto` repo as well to work with this change

so suggest move protobuf change to a new PR

Reply:
The [PR](https://github.com/iotexproject/iotex-proto/pull/110) in iotex-proto has been submitted.